decryptor/mysql/prepared_statement_sql_observer.go

```diff
@@ -0,0 +1,519 @@
+		if columnSetting == nil {
+			logrus.Debugf("No column encryption setting %s", columnName)
+			return nil, false, nil
+		}
+
+		accessContext := base.AccessContextFromContext(ctx)
+		clientID := columnSetting.ClientID()
+		if len(clientID) > 0 {
+			logrus.WithField("client_id", string(clientID)).Debugln("Encrypt with specific ClientID for column")
+		} else {
+			logrus.WithField("client_id", string(accessContext.GetClientID())).Debugln("Encrypt with ClientID from connection")
+			clientID = accessContext.GetClientID()
+		}
+
+		rawData, err := e.coder.Decode(sqlVal, columnSetting)
+		if err != nil {
+			return nil, false, nil
+		}
+
+		encryptedData, err := e.encryptor.EncryptWithClientID(clientID, rawData, columnSetting)
+		if err != nil {
+			return nil, false, err
+		}
+
+		if !bytes.Equal(encryptedData, sqlVal.Val) {
+			output := make([]byte, hex.EncodedLen(len(encryptedData)))
+			hex.Encode(output, encryptedData)
+
+			sqlVal.Val = output
+			sqlVal.Type = sqlparser.HexVal
+			changedRes = true
+		}
+	}
+
+	return base.NewOnQueryObjectFromStatement(setQuery, e.parser), changedRes, nil
+}
+
+func (e *PreparedStatementsQuery) onExecute(ctx context.Context, executeQuery *sqlparser.Execute) (base.OnQueryObject, bool, error) {
+	logrus.Debugln("PreparedStatementsQuery.Execute")
+
+	var preparedStatementName = executeQuery.PreparedStatementName.String()
+
+	_, err := e.proxyHandler.registry.StatementByID(preparedStatementName)
+	if err != nil {
+		logrus.WithField(logging.FieldKeyEventCode, logging.EventCodeErrorGeneral).
+			WithError(err).Errorln("PreparedStatement not present in registry")
+		return nil, false, ErrStatementNotPresentInRegistry
+	}
+
+	querySetting, ok := e.proxyHandler.registry.QuerySettingsByID(preparedStatementName)
+	if ok {
+		logrus.Debugln("Set query settings from registry")
+		e.querySelectSettings = querySetting
+	}
+
+	return base.NewOnQueryObjectFromStatement(executeQuery, e.parser), true, nil
+}
+
+func (e *PreparedStatementsQuery) onDeallocate(ctx context.Context, deallocateQuery *sqlparser.DeallocatePrepare) (base.OnQueryObject, bool, error) {
+	logrus.Debugln("PreparedStatementsQuery.Deallocate")
+
+	var preparedStatementName = deallocateQuery.PreparedStatementName.String()
+
+	if _, err := e.proxyHandler.registry.StatementByID(deallocateQuery.PreparedStatementName.String()); err != nil {
+		logrus.WithField(logging.FieldKeyEventCode, logging.EventCodeErrorGeneral).
+			WithError(err).Errorln("PreparedStatement not present in registry")
+		return nil, false, ErrStatementNotPresentInRegistry
+	}
+
+	e.proxyHandler.registry.DeleteStatementByID(preparedStatementName)
+	return nil, false, nil
+}
+
+func (e *PreparedStatementsQuery) onQuery(ctx context.Context, preparedStatementName string, stmt sqlparser.Statement) (base.OnQueryObject, bool, error) {
+	var query = sqlparser.String(stmt)
+	var preparedStatement = NewPreparedStatement(0, 0, query, stmt)
+
+	preparedStatement.name = preparedStatementName
```

Comment:
same here, let's use new constructor for that to avoid direct assigning to a variable after creation new obj